src/builder.rs

```diff
@@ -6,18 +6,32 @@ use utils::Toggle;
 use params::NoiseParams;
 use resolvers::CryptoResolver;
 use error::{SnowError, InitStage, Prerequisite};
+use subtle::ConstantTimeEq;
 
 /// A keypair object returned by [`generate_keypair()`]
 ///
 /// [`generate_keypair()`]: #method.generate_keypair
-#[derive(PartialEq)]
 pub struct Keypair {
     /// The private asymmetric key
     pub private: Vec<u8>,
     /// The public asymmetric key
     pub public: Vec<u8>,
 }
 
+
+impl PartialEq for Keypair {
+    fn eq(&self, other: &Keypair) -> bool {
+        let priv_eq = self.private.ct_eq(&other.private);
+        let pub_eq = self.public.ct_eq(&other.public);
+
+        if (priv_eq & pub_eq).into() {
+            true
+        } else {
+            false
+        }
```

Comment:
this whole thing can just be `(priv_eq & pub_eq).into()`, the `if {} else {}` isn't necessary.